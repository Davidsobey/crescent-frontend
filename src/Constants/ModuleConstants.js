const ModuleConstants = {
  CREATE_REQUEST: 'CREATE_MODULE_REQUEST',
  CREATE_SUCCESS: 'CREATE_MODULE_SUCCESS',
  CREATE_FAILURE: 'CREATE_MODULE_FAILURE',

  GETALL_REQUEST: 'GETALL_MODULE_REQUEST',
  GETALL_SUCCESS: 'GETALL_MODULE_SUCCESS',
  GETALL_FAILURE: 'GETALL_MODULE_FAILURE',

  UPLOAD_REQUEST: 'UPLOAD_MATERIAL_REQUEST',
  UPLOAD_SUCCESS: 'UPLOAD_MATERIAL_SUCCESS',
  UPLOAD_FAILURE: 'UPLOAD_MATERIAL_FAILURE',

  LOADMODULE_REQUEST: 'LOADMODULE_MODULE_REQUEST',
  LOADMODULE_SUCCESS: 'LOADMODULE_MODULE_SUCCESS',
  LOADMODULE_FAILURE: 'LOADMODULE_MODULE_FAILURE',

  CLEAR_MODULES: 'CLEAR_MODULES',

  MODULE_MATERIAL_REQUEST: 'MODULE_MATERIAL_MODULE_REQUEST',
  MODULE_MATERIAL_SUCCESS: 'MODULE_MATERIAL_MODULE_SUCCESS',
  MODULE_MATERIAL_FAILURE: 'MODULE_MATERIAL_MODULE_FAILURE',
};

export default ModuleConstants;
