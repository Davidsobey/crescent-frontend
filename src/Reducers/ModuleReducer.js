import ModuleConstants from '../Constants/ModuleConstants';
import UserConstants from '../Constants/UserConstants';

function ModuleReducer(state = {}, action) {
  switch (action.type) {
    case ModuleConstants.CREATE_REQUEST:
      return {
        modules: action.moduleName,
      };
    case ModuleConstants.CREATE_SUCCESS:
      return {
        modules: action.moduleName,
      };
    case ModuleConstants.CREATE_FAILURE:
      return {};

    case ModuleConstants.GETALL_REQUEST:
      return {
        loading: true,
      };
    case ModuleConstants.GETALL_SUCCESS:
      return Object.assign({}, state, {
        modules: action.modules,
      });
    case ModuleConstants.GETALL_FAILURE:
      return {
        modules: {},
      };
    case ModuleConstants.LOADMODULE_REQUEST:
      return {
        loading: true,
      };
    case ModuleConstants.LOADMODULE_SUCCESS:
      return Object.assign({}, state, {
        modules: action.modules,
        loading: false,
      });
    case ModuleConstants.LOADMODULE_FAILURE:
      return {
        modules: {},
      };
    case ModuleConstants.CLEAR_MODULES:
      return {
        modules: undefined,
      };
    case ModuleConstants.MODULE_MATERIAL_REQUEST:
      return Object.assign({}, state, {
        loadingMaterial: true,
      });
    case ModuleConstants.MODULE_MATERIAL_SUCCESS:
      return Object.assign({}, state, {
        loadingMaterial: false,
        moduleMaterial: action.moduleMaterials,
      });
    case ModuleConstants.MODULE_MATERIAL_FAILURE:
      return Object.assign({}, state, { moduleMaterial: [] });

    case ModuleConstants.LOADTESTS_REQUEST:
      return Object.assign({}, state, {
        loadingTests: true,
      });
    case ModuleConstants.LOADTESTS_SUCCESS:
      return Object.assign({}, state, {
        loadingTests: false,
        moduleTests: action.moduleTests,
      });
    case ModuleConstants.LOADTESTS_FAILURE:
      return Object.assign({}, state, { moduleTests: [] });
    case UserConstants.LOGOUT:
      return {};
    case ModuleConstants.LOAD_MODULE_REQUEST:
      return state;
    case ModuleConstants.LOAD_MODULE_SUCCESS:
      return Object.assign({}, state, {
        module: action.module,
        loading: false,
      });
    case ModuleConstants.LOAD_MODULE_FAILURE:
      return state;
    default:
      return state;
  }
}

export default ModuleReducer;
