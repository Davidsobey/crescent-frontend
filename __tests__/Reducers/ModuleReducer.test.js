// Reducer to be tested
import ModuleReducer from '../../src/Reducers/ModuleReducer';
import ModuleConstants from '../../src/Constants/ModuleConstants';

describe('ModuleReducer', () => {
  test('CREATE_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.CREATE_SUCCESS, 
      newModuleId: 1,
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETALL_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.GETALL_SUCCESS, 
      modules: [],
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('UPLOAD_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.UPLOAD_SUCCESS, 
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('LOADMODULE_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.LOADMODULE_SUCCESS, 
      modules: [],
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
    
  test('MODULE_MATERIAL_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.MODULE_MATERIAL_SUCCESS, 
      moduleMaterials: [],
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('LOADTESTS_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.LOADTESTS_SUCCESS, 
      moduleTests: [],
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('LOAD_MODULE_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.LOAD_MODULE_SUCCESS, 
      module: {id:1, name:'111'},
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('EDIT_MODULE_SUCCESS', () => {
    const action = { 
      type: ModuleConstants.EDIT_MODULE_SUCCESS, 
    };

    expect(ModuleReducer(undefined, action)).toMatchSnapshot();
  });
});
