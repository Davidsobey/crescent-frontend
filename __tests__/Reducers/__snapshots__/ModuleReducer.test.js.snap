// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleReducer CREATE_SUCCESS 1`] = `
Object {
  "creating": false,
  "newModuleId": 1,
}
`;

exports[`ModuleReducer EDIT_MODULE_SUCCESS 1`] = `
Object {
  "module": undefined,
  "module_editing": false,
}
`;

exports[`ModuleReducer GETALL_SUCCESS 1`] = `
Object {
  "loading": false,
  "modules": Array [],
}
`;

exports[`ModuleReducer LOAD_MODULE_SUCCESS 1`] = `
Object {
  "module": Object {
    "id": 1,
    "name": "111",
  },
  "module_loading": false,
}
`;

exports[`ModuleReducer LOADMODULE_SUCCESS 1`] = `
Object {
  "loading": false,
  "modules": Array [],
}
`;

exports[`ModuleReducer LOADTESTS_SUCCESS 1`] = `
Object {
  "loadingTests": false,
  "moduleTests": Array [],
}
`;

exports[`ModuleReducer MODULE_MATERIAL_SUCCESS 1`] = `
Object {
  "loadingMaterial": false,
  "moduleMaterial": Array [],
}
`;

exports[`ModuleReducer UPLOAD_SUCCESS 1`] = `
Object {
  "uploading": false,
}
`;
